cl/corpus_importer/management/commands/harvard_merge.py

```diff
@@ -0,0 +1,895 @@
+        judges = titlecase(", ".join(find_all_judges(unidecode(harvard_data))))
+
+        if (
+            harvard_clean.issuperset(cl_clean) or cl_data_upper
+        ) and harvard_clean != cl_clean:
+            OpinionCluster.objects.filter(id=cluster_id).update(judges=judges)
+        elif not harvard_clean.intersection(set(find_all_judges(cl_data))):
+            raise JudgeException("Judges are completely different.")
+
+
+def merge_cluster_dates(
+    cluster_id: int,
+    field_name: str,
+    overlapping_data: Optional[Tuple[str | None, date]],
+) -> None:
+    """Compare two dates and choose the best to update the opinion cluster
+    the value if one value is better than the other
+
+    :param cluster_id: Cluster id to update
+    :param field_name: field name to update in opinion cluster
+    :param overlapping_data: data to compare
+    :return: None
+    """
+    if overlapping_data:
+        harvard_data, cl_date = overlapping_data
+        cluster = OpinionCluster.objects.filter(id=cluster_id).first()
+        if harvard_data:
+            harvard_date, harvard_date_is_approximate = validate_dt(
+                harvard_data
+            )
+            if cluster.docket.source == Docket.SCRAPER:
+                # Give preference to harvard data
+                if harvard_date != cl_date:
+                    OpinionCluster.objects.filter(id=cluster_id).update(
+                        **{field_name: harvard_date}
+                    )
+            elif (
+                cluster.date_filed_is_approximate
+                and not harvard_date_is_approximate
+            ):
+                # For some reason docket source is different, then check if
+                # one date is approximate and the other is not if harvard
+                # date is not approximate, it should be better
+                OpinionCluster.objects.filter(id=cluster_id).update(
+                    **{field_name: harvard_date}
+                )
+
+
+def merge_strings(
+    cluster_id: int, field_name: str, overlapping_data: Tuple[str, str]
+) -> None:
+    """Compare two strings and choose the largest
+
+    :param cluster_id: Cluster id to update
+    :param field_name: field name to update in opinion cluster
+    :param overlapping_data: data to compare from harvard and courtlistener
+    :return: None
+    """
+    harvard_data, cl_data = overlapping_data[0], overlapping_data[1]
+    if len(harvard_data) > len(cl_data):
+        OpinionCluster.objects.filter(id=cluster_id).update(
+            **{field_name: harvard_data}
+        )
+
+
+def merge_docket_numbers(cluster_id: int, harvard_docket_number: str) -> None:
+    """Merge Docket Numbers
+
+    :param cluster_id: The cluster id of the merging item
+    :param harvard_docket_number: The harvard docket number
+    :return: None
+    """
+    cl_docket = OpinionCluster.objects.get(id=cluster_id).docket
+
+    if cl_docket.docket_number:
```

Comment:
How come we don't add the docket number if the CL item doesn't start with one?

Reply:
Now we save the harvard docket number if cl docket doesn't have one.